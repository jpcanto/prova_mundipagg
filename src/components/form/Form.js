import React from 'react'
import { bindActionCreators } from 'redux'
import { ChromePicker } from "react-color"
import Files from 'react-files'

export default class Form extends React.Component {

    state = {
        appName: 'App Name',
        appCategory: 'App Category',
        visualizaColorPicker: false,
        mudaCor: '#191919',
        color: {},
        files: [],
        appInfo: {},
    }

    onFilesChange = (files) => {
        if (this.state.files.length < 1) {
            this.setState({
                files
            })
        } else {
            alert('Você já selecionou uma imagem');
        }
    }

    onFilesError = (error, file) => {
        console.log('error code ' + error.code + ': ' + error.message)
    }

    filesRemoveAll = () => {
        this.refs.files.removeFiles()
    }

    mostraColorPicker = () => {
        this.setState({ visualizaColorPicker: true });
    };

    escondeColorPicker = () => {
        this.setState({ visualizaColorPicker: false });
    };

    mudaColorPicker = color => {
        this.setState({ color: color.rgb, mudaCor: color.hex });
    };

    salvaAppInfo = () => {
        if (!this.state.files) {
            alert('É necessário escolher alguma imagem');
        } else if (this.state.mudaCor === '#191919') {
            alert('É necessário escolher alguma cor');
        } else if (!this.state.appCategory === 'App Category') {
            alert('É necessário escolher alguma categoria');
        } else {
            alert(`Dados enviados com sucesso!!`)
            console.log(`Nome: ${this.state.appName}`)
            console.log(`Imagem url: ${this.state.files}`)
            console.log(`Cor hex: ${this.state.mudaCor}`)
            console.log(`Categoria: ${this.state.appCategory}`)
        }
    }

    render() {
        return (
            <form className="conteudo-principal">
                <div className="formulario">
                    <h1>Create Your App</h1>
                    <h2>APP NAME</h2>
                    <input
                        type="text"
                        onChange={(e) => this.setState({ appName: e.target.value })}
                        required
                        placeholder="Enter App Display Name">
                    </input>
                    <h2>APP ICON</h2>
                    <div className="formulario__icone">
                        <p>Select a File</p>
                        <Files
                            ref='files'
                            className='files-dropzone-gallery'
                            onChange={this.onFilesChange}
                            onError={this.onFilesError}
                            accepts={['image/*']}
                            multiple
                            clickable
                        > <div className="files-box">
                                <button className="formulario__icone__select-file">SELECT A FILE</button>
                            </div>
                        </Files>
                    </div>
                    <h2>ICON'S BACKGROUND COLOR</h2>
                    <div className="formulario__cor">
                        <p>Pick a color</p>
                        <button className="formulario__cor__select-color" style={{ background: this.state.mudaCor }} onClick={() => this.mostraColorPicker()}>{this.state.mudaCor}</button>
                        {this.state.visualizaColorPicker && (
                            <div className={"formulario__cor-caixa"}>
                                <div
                                    className={"formulario__cor-caixa--fecha"}
                                    onClick={() => this.escondeColorPicker()}
                                />
                                <ChromePicker
                                    color={this.state.color}
                                    onChange={this.mudaColorPicker}
                                />
                            </div>
                        )}
                    </div>
                    <h2>CATEGORY</h2>
                    <div className="formulario__categoria">
                        <select onChange={(e) => this.setState({ appCategory: e.target.value })}>
                            <option>Select a Category</option>
                            <option>Técnologia e informação</option>
                            <option>Educação</option>
                            <option>Hospitalar</option>
                            <option>Esportes</option>
                        </select>
                    </div>
                </div>
                <div className="preview">
                    <h2>APP'S CARD PREVIEW</h2>
                    <div className="preview__corpo">
                        <div className="preview__corpo--caixa" style={{ background: this.state.mudaCor }}>
                            {
                                this.state.files.length > 0 ?
                                    <div className='files-gallery'>
                                        {this.state.files.map((file) =>
                                            <img className='files-gallery-item' src={file.preview.url} key={file.id} />
                                        )}
                                    </div>
                                    : null
                            }
                        </div>
                        <div className="preview__corpo--descricao">
                            <h2>{this.state.appName}</h2>
                            <p>{this.state.appCategory}</p>
                            <p className="small">New App</p>
                        </div>
                    </div>
                    <button onClick={this.salvaAppInfo}>SAVE APP</button>
                </div>
            </form>
        )
    }
}
