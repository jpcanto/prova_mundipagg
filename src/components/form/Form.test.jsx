import React from 'react';
import Form from './Form.js';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe('Testando renderização do componente Form', () => {
    it("Deve renderizar corretamente", () => {

        const wrapper = mount(
            <Form />
        );

        expect(toJson(wrapper)).toMatchSnapshot()

    });
});

describe('Testando chamada do color picker', () => {
    it("Deve abrir o color picker ao clicar no botão", () => {

        const wrapper = mount(
            <Form />
        );

        expect(wrapper.state('visualizaColorPicker')).toBe(false);
        wrapper.find('.formulario__cor__select-color').simulate('click');
        expect(wrapper.state('visualizaColorPicker')).toBe(true)

    });

    it("Deve fechar o color picker ao clicar fora da caixa", () => {

        const wrapper = mount(
            <Form />
        );

        wrapper.setState({
            visualizaColorPicker: true
        })

        expect(wrapper.state('visualizaColorPicker')).toBe(true);
        wrapper.find('.formulario__cor-caixa--fecha').simulate('click');
        expect(wrapper.state('visualizaColorPicker')).toBe(false)

    });
});