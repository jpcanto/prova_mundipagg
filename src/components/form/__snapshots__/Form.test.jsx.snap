// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando renderização do componente Form Deve renderizar corretamente 1`] = `
<Form>
  <section
    className="conteudo-principal"
  >
    <div
      className="formulario"
    >
      <h1>
        Create Your App
      </h1>
      <button
        className="formulario__fecha"
        onClick={[Function]}
      >
        X
      </button>
      <h2>
        APP NAME
      </h2>
      <input
        onChange={[Function]}
        placeholder="Enter App Display Name"
        type="text"
      />
      <h2>
        APP ICON
      </h2>
      <div
        className="formulario__icone"
      >
        <t
          accepts={
            Array [
              "image/*",
            ]
          }
          className="files-dropzone"
          clickable={true}
          dropActiveClassName="files-dropzone-active"
          maxFileSize={Infinity}
          maxFiles={Infinity}
          minFileSize={0}
          multiple={false}
          name="file"
          onChange={[Function]}
          onError={[Function]}
        >
          <div>
            <input
              accept="image/*"
              multiple={false}
              name="file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <div
              className="files-dropzone"
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
               
              <div
                className="formulario__icone__select-box"
              >
                <p>
                  Select a File
                </p>
                <button
                  className="formulario__icone__select-box__file"
                >
                  SELECT A FILE
                </button>
              </div>
            </div>
          </div>
        </t>
      </div>
      <h2>
        ICON'S BACKGROUND COLOR
      </h2>
      <div
        className="formulario__cor"
      >
        <p
          style={null}
        >
          Pick a color
        </p>
        <button
          className="formulario__cor__select-color"
          onClick={[Function]}
          style={
            Object {
              "background": "#191919",
            }
          }
        >
          #191919
        </button>
      </div>
      <h2>
        CATEGORY
      </h2>
      <div
        className="formulario__categoria"
      >
        <select
          onChange={[Function]}
        >
          <option>
            Select a Category
          </option>
          <option>
            Técnologia e informação
          </option>
          <option>
            Educação
          </option>
          <option>
            Hospitalar
          </option>
          <option>
            Esportes
          </option>
        </select>
      </div>
    </div>
    <div
      className="preview"
    >
      <h2>
        APP'S CARD PREVIEW
      </h2>
      <div
        className="preview__corpo"
      >
        <div
          className="preview__corpo--caixa"
          style={
            Object {
              "background": "#191919",
            }
          }
        >
          <img
            alt="img default"
            className="files-gallery-item"
            src="https://cdn2.iconfinder.com/data/icons/status-solid-style/24/close-circle-512.png"
          />
        </div>
        <div
          className="preview__corpo--descricao"
        >
          <h2>
            App Name
          </h2>
          <p>
            App Category
          </p>
          <p
            className="small"
          >
            New App
          </p>
        </div>
      </div>
      <button
        onClick={[Function]}
      >
        SAVE APP
      </button>
    </div>
  </section>
</Form>
`;
